### ISSUE TITLE
Ability to add Identity Provider sign in buttons to the register.ftl page

### ISSUE THREAD

**Issue by gordody** [FusionAuth/fusionauth-issues#554]
### Problem
Our current design calls for the ability to sign in using Identity Providers from the register.ftl page (besides the authorize.ftl page). This is currently not possible: even after adding the same imports to register.ftl as I have on authorize.ftl the `identityProviders` variables are empty on the register.ftl page.

Reasoning: Sign-up and sign-in are very similar using an identity provider, so if you have an identity provider account you click the Log in button and you automatically get added to the user database.

### Solution
Please add the `identityProviders` structure to the registrater.ftl (and possibly the other pages as well, just in case)

### Alternatives/workarounds
Copy the necessary data from authorize.ftl and manually add it into register.ftl

### Additional context
Sign up design:
![image](https://user-images.githubusercontent.com/24533582/79006042-1bd5a900-7b0d-11ea-9314-cb8220440896.png)

### Related
- https://github.com/FusionAuth/fusionauth-issues/issues/816

### How to vote
Please give us a thumbs up or thumbs down as a reaction to help us prioritize this feature. Feel free to comment if you have a particular need or comment on how this feature should work.

**Comment by ulybu** [comment]
**Additional arguments for this needs:** 
Having Register being more prominent than Login is a common pattern in the industry, particularly when you want to encourage new users or if logouts aren't frequent. In that case, we need to be able to point directly to a full Register page, including social providers.

FusionAuth already allows to customize per app if a social provider is to be used for login only or also registration. So the register page could show only those social provider which are allowed for registration without risk. 

![image](https://user-images.githubusercontent.com/2479249/83253561-75716000-a1ad-11ea-9e05-d9a627e0a81f.png)

@gordody Could you elaborate on that workaround? I couldn't make it work :/

**Comment by robotdan** [comment]
@ulybu thanks - that is a good point. We could optionally limit the providers to those configured for registration. 

Something like :

````
  Register for Acme Corp.
  ------------------------

   email: ________
   password: ________
   confirm: ________

   -- or register using --

   [ Facebook ]
   [ Google ]


  ------------------------
  Already have an account? Login in here. (goes to normal login page)
````

**Comment by ulybu** [comment]
> Something like :

Exactly what I had in mind ;)

**Comment by gordody** [comment]
@ulybu I basically copied the code from the generated content into the signup template, you can check it out in action on the signup page of design.famous.co

Here is my code from oauth2Register.ftl:

```
...
    [@helpers.header]
      [#-- Custom header code goes here --]
    [/@helpers.header]

    [@helpers.main title='']
    [#-- TODO: uncomment when we have identity providers
      [@helpers.alternativeLogins clientId=client_id identityProviders=identityProviders passwordlessEnabled=passwordlessEnabled/]
    --]
      [#-- TODO: delete when there are real identity providers --]
      <div class="login-button-container">
        <div class="form-row push-less-top">
          <button id="google-login-button" class="google famous-identity-provider-login-button" data-scope="">
            <div>
              <div class="icon">
                <svg version="1.1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48" width="25" height="25">
                  <g>
                    <path class="cls-1" d="M24 9.5c3.54 0 6.71 1.22 9.21 3.6l6.85-6.85C35.9 2.38 30.47 0 24 0 14.62 0 6.51 5.38 2.56 13.22l7.98 6.19C12.43 13.72 17.74 9.5 24 9.5z"></path>
                    <path class="cls-2" d="M46.98 24.55c0-1.57-.15-3.09-.38-4.55H24v9.02h12.94c-.58 2.96-2.26 5.48-4.78 7.18l7.73 6c4.51-4.18 7.09-10.36 7.09-17.65z"></path>
                    <path class="cls-3" d="M10.53 28.59c-.48-1.45-.76-2.99-.76-4.59s.27-3.14.76-4.59l-7.98-6.19C.92 16.46 0 20.12 0 24c0 3.88.92 7.54 2.56 10.78l7.97-6.19z"></path>
                    <path class="cls-4" d="M24 48c6.48 0 11.93-2.13 15.89-5.81l-7.73-6c-2.15 1.45-4.92 2.3-8.16 2.3-6.26 0-11.57-4.22-13.47-9.91l-7.98 6.19C6.51 42.62 14.62 48 24 48z"></path>
                    <path class="cls-5" d="M0 0h48v48H0z"></path>
                  </g>
                </svg>
              </div>
              <div class="text">Sign in with Google</div>
            </div>
          </button>
        </div>
        <div class="form-row push-less-top">
          <button id="facebook-login-button" class="facebook famous-identity-provider-login-button" data-permissions="public_profile, email">
            <div>
              <div class="icon">
                <svg version="1.1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 216 216" width="25" height="25">
                  <path class="cls-1" d="M204.1 0H11.9C5.3 0 0 5.3 0 11.9v192.2c0 6.6 5.3 11.9 11.9 11.9h103.5v-83.6H87.2V99.8h28.1v-24c0-27.9 17-43.1 41.9-43.1 11.9 0 22.2.9 25.2 1.3v29.2h-17.3c-13.5 0-16.2 6.4-16.2 15.9v20.8h32.3l-4.2 32.6h-28V216h55c6.6 0 11.9-5.3 11.9-11.9V11.9C216 5.3 210.7 0 204.1 0z"></path>
                </svg>
              </div>
              <div class="text">Sign in with Facebook</div>
            </div>
          </button>
        </div>

        <div class="famous-hr-container">
          <div>Or</div>
        </div>

      </div>
      [#-- END TODO: delete when there are real identity providers --]
...
```

**Comment by godguide1** [comment]
Thanks @robotdan for the quick response.

We built our custom registration and login page using Vue js, and we just called the register and login API. We also want to add, on these pages, Identity Providers button. My question is, is this something we can achieve on our Vue.js custom pages? 

![signup](https://user-images.githubusercontent.com/37475078/85340725-f52ee800-b529-11ea-840d-2676ca1c6598.PNG)

![signin](https://user-images.githubusercontent.com/37475078/85340750-ffe97d00-b529-11ea-9616-d0fdae04f3bc.PNG)

?

**Comment by robotdan** [comment]
If you aren't using our themed login pages, but building your own registration workflow using our APIs, then yes, you could build this yourself. 

You'd use the Identity Provider APIs to complete the Google, Facebook or Twitter logins after retrieving a token from the identity provider. 
https://fusionauth.io/docs/v1/tech/apis/identity-providers/

**Comment by robotdan** [comment]
This is available in 1.28.0, once documentation is updated, or confirmed to be updated we'll close this one out.
